apis/low_level_debug/src/lib.rs

```diff
@@ -0,0 +1,73 @@
+#![no_std]
+
+use libtock_platform::{CommandReturn, Syscalls};
+
+/// The low-level debug API provides tools to diagnose userspace issues that make normal debugging workflows (e.g. printing to the console) difficult.
+///
+/// It allows libraries to print alert codes and apps to print numeric information using only the command system call.
+///
+/// # Example
+/// ```ignore
+/// use libtock_runtime::TockSyscalls;
+///
+/// // Uses the real syscall implementation.
+/// type LowLevelDebug = libtock_low_level_debug::LowLevelDebug<TockSyscalls>;
+///
+/// // Prints 0x45 and the app which called it.
+/// LowLevelDebug::print_1(0x45);
+/// ```
+pub struct LowLevelDebug<S: Syscalls>(S);
+
+impl<S: Syscalls> LowLevelDebug<S> {
+    /// Run a check against the low-level debug capsule to ensure it is working.
```

Comment:
Should probably say "present" rather than "working". The failure mode here is if the kernel does not have the `LowLevelDebug` capsule.

Reply:
Perhaps, though it seems like an implementation detail that "check if it's working" only does a presence check.

Reply:
`LowLevelDebug` can "fail" if it is unable to allocate grant memory. If that occurs, it will try to print out a "grant init failed" message, but regardless it won't work correctly. The driver check command will return Success in that case regardless.

As-is, this comment implies that if `driver_check` returns Success then LowLevelDebug is working, which is not necessarily true.

Reply:
That's not really a "driver check" then, is it? It seems like a bug in the `LowLevelDebug` capsule.

Reply:
I updated the comment, though I would like us to consider having it make an actual check in the future.

Reply:
I think this discussion is worth continuing upstream, so I opened https://github.com/tock/tock/pull/2908.